### PR TITLE
feat(design): Add Variants to List Molecule

### PR DESCRIPTION
**Pull request by xelaint**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

File: libs/design/src/molecules/list/list-subheader/list-subheader.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { DaffListSubheaderComponent } from './list-subheader.component';
+
+describe('DaffListSubheaderComponent', () => {
+  it('should create an instance', () => {
+    const component = new DaffListSubheaderComponent();
+    expect(component).toBeTruthy();
+  });
+});
```
Comment (damienwebdev) [inline libs/design/src/molecules/list/list-subheader/list-subheader.component.spec.ts]: Needs a test about adding css class to host.

**Comment by xelaint** [comment]
> I don't think link-list should be a part of this component. I think it should be a separate component.

@damienwebdev that's what I was kind of thinking when I added it. I'll remove it from this component.

File: libs/design/src/molecules/list/list-subheader/list-subheader.component.spec.ts (hunk 1 above)
Comment (xelaint) [reply libs/design/src/molecules/list/list-subheader/list-subheader.component.spec.ts]: Resolved in 2ba76af

File: apps/foundation-demo/src/app/core/footer/footer.component.spec.ts [hunk 2]
```diff
@@ -38,4 +42,12 @@ describe('FooterComponent', () => {
       expect(container.componentInstance.size).toEqual('medium');
     });
   });
+
+  describe('on all <daff-list>', () => {
```
Comment (damienwebdev) [inline apps/foundation-demo/src/app/core/footer/footer.component.spec.ts]: I think this is a poor test. Let's reimagine this driven from a requirement:

```
describe('footer')...
describe('the list of navigational links')

it('should render three daff-lists of type `link`')
```

File: libs/design/src/molecules/list/list/list.component.scss [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  &__subheader {
+    @include embolden();
+    @include uppercase();
+    font-size: 1rem;
+    padding: 0 0 25px 0;
+  }
+
+  &__item {
+    display: block;
+    padding: 8px 0;
```
Comment (damienwebdev) [inline libs/design/src/molecules/list/list/list.component.scss]: Margin, not padding.

File: libs/design/src/molecules/list/list/list.component.scss [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+    &:first-of-type {
+      padding: 0 0 8px 0;
```
Comment (damienwebdev) [inline libs/design/src/molecules/list/list/list.component.scss]: Margin, not padding.

File: libs/design/src/molecules/list/list/list.component.scss (hunk 3 above)
Comment (xelaint) [reply libs/design/src/molecules/list/list/list.component.scss]: I disagree. What if I want to add a background color hover effect on a `list-item`?

File: libs/design/src/molecules/list/list/list.component.scss (hunk 4 above)
Comment (xelaint) [reply libs/design/src/molecules/list/list/list.component.scss]: > I disagree. What if I want to add a background color hover effect on a `list-item`?

File: apps/foundation-demo/src/app/core/footer/footer.component.spec.ts (hunk 2 above)
Comment (xelaint) [reply apps/foundation-demo/src/app/core/footer/footer.component.spec.ts]: Resolved in df1d786

File: libs/design/src/molecules/list/list/list.component.scss (hunk 3 above)
Comment (damienwebdev) [reply libs/design/src/molecules/list/list/list.component.scss]: Hrm. Then I don't think that [daff-list-item] should be an attribute. It should be a element. We don't style the content of list items in lists, we merely style the element and a few distinct elements like subheader.

File: apps/foundation-demo/src/app/core/footer/footer.component.spec.ts [hunk 5]
```diff
@@ -38,4 +42,14 @@ describe('FooterComponent', () => {
       expect(container.componentInstance.size).toEqual('medium');
     });
   });
+
+  describe('footer', () => {
+    describe('the list of navigational links', () => {
+      it('should render three daff-lists with type="link"', () => {
```
Comment (damienwebdev) [inline apps/foundation-demo/src/app/core/footer/footer.component.spec.ts]: This test is wrong still.

**Pull request merged**